packages/entity/src/EntityFieldDefinition.ts

```diff
@@ -93,6 +122,14 @@ export interface EntityAssociationDefinition<
    *   integrity is recommended.
    */
   edgeDeletionBehavior: EntityEdgeDeletionBehavior;
+
+  /**
+   * Optimization setting for evaluation of this edge in canViewerDeleteAsync. Can be used to optimize permission checks
+   * for edge cascading deletions based on application-specific design of the entity association. Not used during actual deletions.
+   *
+   * Defaults to EntityEdgeDeletionAuthorizationInferenceBehavior.NONE.
+   */
+  edgeDeletionAuthorizationInferenceBehavior?: EntityEdgeDeletionAuthorizationInferenceBehavior;
```

Comment:
Should this be required?
- Pros: it'd be harder for application developers to forget to add it when applicable.
- Cons: it's somewhat hard to understand as a concept, would require adding to all associations. Unnecessary for applications not using `canViewerDeleteAsync`.

```suggestion
  edgeDeletionAuthorizationInferenceBehavior: EntityEdgeDeletionAuthorizationInferenceBehavior;
```

Reply:
My gut feeling is to make it optional since it is an advanced concept and an optimization that makes sense only for large fanouts. In contrast, someone adding a 1:1 edge wouldn't need to implement this at all.